r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/MySqlSimpleConnection.java

```diff
@@ -563,7 +563,7 @@ private static Mono<SessionData> loadSessionVariables(Client client, Codecs code
     }
 
     private static Mono<SessionData> loadInnoDbEngineStatus(SessionData data, Client client, Codecs codecs) {
-        return new TextSimpleStatement(client, codecs, "SHOW VARIABLES LIKE 'innodb\\\\_lock\\\\_wait\\\\_timeout'")
+        return new TextSimpleStatement(client, codecs, "SHOW VARIABLES LIKE 'innodb\\_lock\\_wait\\_timeout'")
```

Comment:
it seems this query won't work if `isNoBackslashEscapes() == true` :(

Reply:
I tested, and it should work on `mysql:5.5`, `mariadb:latest` and `mysql:latest`.

See the following:

```sql
SET sql_mode = 'STRICT_TRANS_TABLES';

SHOW VARIABLES LIKE 'innodb\_lock\_wait\_timeout'; -- 1 row
SHOW VARIABLES LIKE 'innodb\\_lock\\_wait\\_timeout'; -- 1 row

SET sql_mode = 'STRICT_TRANS_TABLES,NO_BACKSLASH_ESCAPES';

SHOW VARIABLES LIKE 'innodb\_lock\_wait\_timeout'; -- 1 row
SHOW VARIABLES LIKE 'innodb\\_lock\\_wait\\_timeout'; -- 0 row
```

If `NO_BACKSLASH_ESCAPES` is set, `\_` will be equivalent to `\\_` on `NO_BACKSLASH_ESCAPES` is unset. The backslash is used by `LIKE` matching instead of escaping.

If `NO_BACKSLASH_ESCAPES` is not set, the `\_` escape will be ignored, and the original string `\_` will be obtained by matching.

Reply:
could you share how do you test it? I've used MySQL 8.3.0 Community edition(mysql:latest docker tag) and mysql client (`mysql  Ver 8.2.0 for macos14.0 on arm64 (Homebrew)`) and I've got below result.

```
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 9
Server version: 8.3.0 MySQL Community Server - GPL

Copyright (c) 2000, 2023, Oracle and/or its affiliates.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> SET sql_mode = 'STRICT_TRANS_TABLES';
Query OK, 0 rows affected, 1 warning (0.00 sec)

mysql> 
mysql> SHOW VARIABLES LIKE 'innodb\_lock\_wait\_timeout'; -- 1 row
+--------------------------+-------+
| Variable_name            | Value |
+--------------------------+-------+
| innodb_lock_wait_timeout | 50    |
+--------------------------+-------+
1 row in set (0.01 sec)

mysql> SHOW VARIABLES LIKE 'innodb\\_lock\\_wait\\_timeout'; -- 1 row
+--------------------------+-------+
| Variable_name            | Value |
+--------------------------+-------+
| innodb_lock_wait_timeout | 50    |
+--------------------------+-------+
1 row in set (0.01 sec)

mysql> 
mysql> SET sql_mode = 'STRICT_TRANS_TABLES,NO_BACKSLASH_ESCAPES';
Query OK, 0 rows affected, 1 warning (0.00 sec)

mysql> 
mysql> SHOW VARIABLES LIKE 'innodb\_lock\_wait\_timeout'; -- 1 row
Empty set (0.00 sec)

mysql> SHOW VARIABLES LIKE 'innodb\\_lock\\_wait\\_timeout'; -- 0 row
Empty set (0.01 sec)
```

Reply:
@mirromutth 
Understood. Let's proceed according to the specifications. It seems that My local environment is acting strangely.(https://dev.mysql.com/doc/refman/8.0/en/string-comparison-functions.html#operator_like)

Reply:
@jchrys You're right, it's my fault. For some reason in other projects, I have a 5.7-compatible internal distribution marked as `mysql:latest` on my local environment.

I tried again in a clean environment and found that in MySQL 8.0 and later, this behavior is broken. MySQL 5.7 and MariaDB 11.2.2 can match this pattern, MySQL 8.0 and above cannot.

Reply:
It is changed to `SHOW VARIABLES WHERE Variable_name = 'innodb_lock_wait_timeout'`

Reply:
I think need to report this issue D